src/test/java/com/ait/lienzo/client/core/shape/wires/handlers/impl/WiresConnectorControlPointBuilderTest.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ait.lienzo.client.core.shape.wires.handlers.impl;
+
+import com.ait.lienzo.client.core.animation.IAnimationHandle;
+import com.ait.lienzo.client.core.shape.*;
+import com.ait.lienzo.client.core.shape.wires.WiresConnector;
+import com.ait.lienzo.client.core.shape.wires.decorator.IShapeDecorator;
+import com.ait.lienzo.client.core.shape.wires.decorator.PointHandleDecorator;
+import com.ait.lienzo.client.core.types.Point2D;
+import com.ait.lienzo.client.core.types.Transform;
+import com.ait.lienzo.shared.core.types.ColorName;
+import com.ait.lienzo.test.LienzoMockitoTestRunner;
+import com.ait.tooling.common.api.java.util.function.Function;
+import com.ait.tooling.common.api.java.util.function.Predicate;
+import com.ait.tooling.nativetools.client.event.HandlerRegistrationManager;
+import com.google.gwt.event.shared.HandlerRegistration;
+import com.google.gwt.user.client.Timer;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+
+import static org.junit.Assert.*;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.*;
+
+@RunWith(LienzoMockitoTestRunner.class)
+public class WiresConnectorControlPointBuilderTest
+{
+    private static final MultiPath headPath = spy(new MultiPath().circle(10));
+
+    private static final MultiPath tailPath = new MultiPath().circle(10);
+
+    private static Predicate<WiresConnector> TRUE_PREDICATE = new Predicate<WiresConnector>()
+    {
+        @Override
+        public boolean test(WiresConnector connector)
+        {
+            return true;
+        }
+    };
+
+    @Mock
+    private WiresConnectorControlImpl             control;
+
+    @Mock
+    private PointHandleDecorator              pointHandleDecorator;
+
+    @Mock
+    private HandlerRegistrationManager        controlEvents;
+
+    @Mock
+    private Viewport                          viewport;
+
+    private WiresConnectorControlPointBuilder tested;
+
+    private Layer                             layer;
+
+    private Layer                             overLayer;
+
+    private Transform                         transform;
+
+    private WiresConnector                    connector;
+
+    private IDirectionalMultiPointShape       line;
```

Comment:
`transform`, `connector` and `line` can be local variables.

Reply:
fixed